src/api/index.ts

```diff
@@ -105,54 +106,68 @@ export async function getWorkflowId(workflowFilename: string): Promise<number> {
+      branchName = getBranchNameFromRef(config.ref)
+
+      if (!config.workflow) {
+        throw new Error(`An input to 'workflow' was not provided`)
+      }
+      // https://docs.github.com/en/rest/actions/workflow-runs#list-workflow-runs-for-a-workflow
+      response = await octokit.paginate(
+        octokit.rest.actions.listWorkflowRuns,
+        {
+          owner: config.owner,
+          repo: config.repo,
+          created: `>${startTime.toISOString()}`,
+          workflow_id: config.workflow,
+          ...(branchName ? {branch: branchName} : {})
+        },
+        resp => {
+          core.debug(`Fetched page: ${JSON.stringify(resp, null, 2)}`)
+          return resp.data
+        }
+      )
+    } else {
+      // repository_dipsatch can only be triggered from the default branch
+      const branchName = await getDefaultBranch()
+      // https://docs.github.com/en/rest/actions/workflow-runs#list-workflow-runs-for-a-repository
+      response = await octokit.paginate(
+        octokit.rest.actions.listWorkflowRunsForRepo,
+        {
+          owner: config.owner,
+          repo: config.repo,
+          branch: branchName,
+          created: `>${startTime.toISOString()}`,
+          event: DispatchMethod.RepositoryDispatch
+        },
+        resp => {
+          core.debug(`Fetched page: ${JSON.stringify(resp, null, 2)}`)
+          return resp.data
+        }
+      )
+    }
+  } catch (error) {
+    const err = `Failed to get workflow runs.`
 
-    if (!config.workflow) {
-      throw new Error(`An input to 'workflow' was not provided`)
+    if (error instanceof RequestError) {
+      throw new Error(`${err} Expected 200 but received ${error.status}`)
     }
-    // https://docs.github.com/en/rest/actions/workflow-runs#list-workflow-runs-for-a-workflow
-    response = await octokit.rest.actions.listWorkflowRuns({
-      owner: config.owner,
-      repo: config.repo,
-      workflow_id: config.workflow,
-      ...(branchName
-        ? {
-            branch: branchName,
-            per_page: 5
-          }
-        : {
-            per_page: 10
-          })
-    })
-    status = response.status
-  } else {
-    // repository_dipsatch can only be triggered from the default branch
-    const branchName = await getDefaultBranch()
-    // https://docs.github.com/en/rest/actions/workflow-runs#list-workflow-runs-for-a-repository
-    response = await octokit.rest.actions.listWorkflowRunsForRepo({
-      owner: config.owner,
-      repo: config.repo,
-      branch: branchName,
-      event: DispatchMethod.RepositoryDispatch,
-      per_page: 5
-    })
-    status = response.status
-  }
 
-  if (status !== 200) {
-    throw new Error(
-      `getWorkflowRuns: Failed to get workflow runs, expected 200 but received ${status}`
-    )
+    throw new Error(`${err} ${error}`)
   }
 
-  const workflowRuns: WorkflowRun[] = response.data.workflow_runs.map(
-    workflowRun => ({
```

Comment:
This is required because of the poor usage of type annotations for the paginated responses. Maybe there's something we can do better here.